### DISCUSSION TITLE
Embed Assets

### DISCUSSION THREAD

**Discussion by wwderw** [webui-dev/webui#333] [Ideas]
I was wondering if there is plans to be able to load embedded assets via some function like LoadAssetFromMemory(data,size);?  Right now, I can use base64 for non SVG type of assets (which can have some issues depending on browser and size of the file), but being able to load from an embedded file, using LD to link the assets into the binary and load from there for instance, would help get around that while still being able to make it simple for distribution.

**Comment by AlbertShown** [comment]
Do you mean something like a custom file handler?

https://github.com/webui-dev/webui/blob/55e38e89cec3fee18432e19ff4401c9e23c2ef36/examples/C/serve_a_folder/main.c#L58

**Comment by wwderw** [comment]
Not really, if I'm reading that example correctly.  It looks like it only deals with text files.  This would be more like for images, audio, other assets and being able to load them in that way without having to deal with overhead of converting to header files or base64.

**Comment by AlbertShown** [comment]
Not only text files, but all HTTP requested files should go trough `my_files_handler`, including images, videos, and everything.
Did you test the example and only shows text files?

**Reply by d33p-dev** [reply]
Very cool...  I'll be testing this over next few days.  I think this is the way to handle complex JS/SPA type HTML apps loaded by/into WebUI...  At least, it seems like it is.  What I don't know yet is whether I can manage to get the webUI browser page or webView to load assets that then make API calls and need to control their CORS origin when making remote requests.

**Comment by hassandraga** [comment]
Implemented.
https://github.com/webui-dev/webui/tree/main/examples/C/virtual_file_system

**Reply by d33p-dev** [reply]
This is interesting!  👍
